src/test/java/io/fabric8/vertx/maven/plugin/it/invoker/ControlledVerifier.java

```diff
@@ -0,0 +1,111 @@
+    public void executeGoals(List<String> goals, Map<String, String> envVars) throws VerificationException {
+        DefaultInvocationRequest request = new DefaultInvocationRequest();
+        ControlledInvoker invoker = new ControlledInvoker();
+
+        List<String> allGoals = new ArrayList<>();
+
+        if (isAutoclean()) {
+            allGoals.add("org.apache.maven.plugins:maven-clean-plugin:clean");
+        }
+
+        allGoals.addAll(goals);
+
+        request.setGoals(allGoals);
+
+        File logFile = new File(getBasedir(), getLogFileName());
+
+
+        request.setDebug(isMavenDebug());
+
+        /*
+         * NOTE: Unless explicitly requested by the caller, the forked builds should use the current local
+         * repository. Otherwise, the forked builds would in principle leave the sandbox environment which has been
+         * setup for the current build. In particular, using "maven.repo.local" will make sure the forked builds use
+         * the same local repo as the parent build even if a custom user settings is provided.
+         */
+        boolean useMavenRepoLocal = Boolean.valueOf(getVerifierProperties().getProperty("use.mavenRepoLocal", "true"));
+
+        if (useMavenRepoLocal) {
+            request.setLocalRepositoryDirectory(new File(getLocalRepository()));
+            invoker.setLocalRepositoryDirectory(new File(getLocalRepository()));
+        }
+
+        try {
+//            MavenProcessLauncher launcher =
+//                new MavenProcessLauncher( defaultMavenHome, envVars, isDebugJvm() );
+
+            request.setBaseDirectory(new File(getBasedir()));
+            request.setPomFile(new File(getBasedir(), "pom.xml"));
+
+
+            PrintStream log = new PrintStream(logFile);
+            invoker.setErrorHandler(new PrintStreamHandler(log, true));
+            invoker.setOutputHandler(new PrintStreamHandler(log, true));
+            invoker.setLogger(new PrintStreamLogger(log, InvokerLogger.INFO));
+            findDefaultMavenHome();
+            invoker.setMavenHome(new File(defaultMavenHome));
+            invoker.setWorkingDirectory(new File(getBasedir()));
+            request.setOutputHandler(new PrintStreamHandler(log, true));
+
+            result = (ControlledInvocationResult) invoker.execute(request);
+        } catch (Exception e) {
+            throw new VerificationException(e);
+        }
+    }
+
+    private void findDefaultMavenHome()
+        throws VerificationException {
+        defaultMavenHome = System.getProperty("maven.home");
+
+        if (defaultMavenHome == null) {
+            Properties envVars = CommandLineUtils.getSystemEnvVars();
+            defaultMavenHome = envVars.getProperty("M2_HOME");
+        }
+
+        if (defaultMavenHome == null) {
+            File f = new File(System.getProperty("user.home"), "m2");
+            if (new File(f, "bin/mvn").isFile()) {
+                defaultMavenHome = f.getAbsolutePath();
+            }
+        }
+    }
+}
```

Comment:
just a thought if we can use mvnWrapper if we dont find maven anywhere ? why cant we base all our tests to use mvnWrapper - so that we can stay close to the version are targetting and add the same via maven enforcer plugin?